buildSrc/src/main/kotlin/plugin/LibraryPlugin.kt

```diff
@@ -0,0 +1,56 @@
+package plugin
+
+import com.android.build.gradle.LibraryExtension
+import org.gradle.api.JavaVersion
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.dependencies
+
+class LibraryPlugin : Plugin<Project> {
+
+    override fun apply(project: Project) {
+        applyPlugins(project)
+
+        applyLibraryExtension(project)
+
+        importExternalLibs(project)
+    }
+
+    private fun applyPlugins(project: Project) {
+        project.plugins.apply {
+            apply("com.android.library")
+        }
+    }
+
+    private fun applyLibraryExtension(project: Project) {
+        val extension = project.extensions.getByName("android")
+                as? LibraryExtension ?: return
+        extension.apply {
+            compileSdkVersion(AppConfig.compileSdkVersion)
```

Comment:
This is deprecated. Switch to the most recent DSL. Also remove other uses of deprecated APIs